### DISCUSSION TITLE
Draw molecule from inchi Java

### DISCUSSION THREAD

**Discussion by tnguy271** [rdkit/rdkit#5645] [Q&A] [answered]
Can anyone help me how to create a mol from an inchi passed as a string? I only see MolToInchi method in the API. Thank you so much !

**Comment by greglandrum** [comment, marked as answer]
@tnguy271 it's almost always a bad idea to convert an InChI back to a molecule (InChI was not designed to do this), but if you really must, there is a function called `InChIToMol()` in the Java wrappers

**Reply by dmyersturnbull** [reply]
@greglandrum Sorry to resurrect this thread! I'm wondering about something that might be relevant to others:

Why is converting InChI back to a Mol not recommended? I had always assumed this was fine. (I also can't think of how else InChI can be useful, aside from generating InChI Keys.)

**Reply by DavidACosgrove** [reply]
That is the purpose of an InChi - a one-way encoding that gives a unique code for a molecule enabling exact matches between molecules to be done rapidly and correctly.  It's a hash code for molecules.

**Reply by bp-kelley** [reply]
@DavidACosgrove that's the InchiKey, I believe the question was about the Inchi itself.  

In general, the Inchi is used for registration purposes in that it transforms the molecule from the input using various "layers" such as tautomerization.  If you were protonating a molecule for, say, docking or dynamics analysis, the inchi would most likely remove what you specified.

From their spec:  ""designed to provide a standard way to encode molecular information and to facilitate the search for such information in databases and on the web."""
